XYZDriversWebService/src/main/java/com/webservices/xyzdriverswebservice/ClaimEligibility.java

```diff
@@ -0,0 +1,131 @@
+    public String Eligibility(
+            @WebParam(name = "username") String username,
+            @WebParam(name = "joinedDate") String joinedDate,
+            @WebParam(name = "listOfClaimDates") ArrayList<String> listOfClaimDates,
+            @WebParam(name = "listOfClaimStatuses") ArrayList<String> listOfClaimStatuses) {
+
+        Calendar calendarSixMonths = getSixMonthsAgo();  //set to six months ago
+        Calendar calendarStartYear = getStartYear();  //set to start of current year
+        Calendar calendarEndYear = getEndYear();    //set to end of current year
+        Calendar calendarJoinedDate = Calendar.getInstance(); //incoming joined date is converted to calender
+
+        List<Calendar> calendarClaimDates = new ArrayList();  //incoming List<String> is converted to List<Calender>
+
+        String returnStatement = "";
+
+        try {
+            if (username == null || username.isEmpty()) {
+                throw new IllegalArgumentException("Username is null or empty: " + username);
+            }
+            if (joinedDate == null) {
+                throw new IllegalArgumentException("Joined Date is null: " + joinedDate);
+            }
+            if (listOfClaimDates == null) {
+                throw new IllegalArgumentException("List of claim dates is null: " + listOfClaimDates);
+            }
+
+            if (listOfClaimStatuses == null) {
+                throw new IllegalArgumentException("List of claim statuses is null: " + listOfClaimStatuses);
+            }
+
+            SimpleDateFormat sdf = new SimpleDateFormat("yyyy-MM-dd");
+            Date date = sdf.parse(joinedDate);
+            calendarJoinedDate.setTime(date);
+
+            for (int i = 0; i < listOfClaimDates.size(); i++) //convert dates list
+            {
+                Calendar tempCalendar = Calendar.getInstance();
+                Date tempDate = sdf.parse(listOfClaimDates.get(i));
+                tempCalendar.setTime(tempDate);
+                calendarClaimDates.add(tempCalendar);
+            }
+
+            if (calendarJoinedDate.after(calendarSixMonths)) {
+                returnStatement = " has joined less than six months ago. As such, they are not yet eligible to make a claim.";
+            } else {
+                returnStatement = checkNumberOfClaims(calendarStartYear, calendarEndYear, calendarClaimDates, listOfClaimStatuses);
+            }
+
+        } catch (IllegalArgumentException | ParseException ex) {
+            return ex.toString();
+        }
+
+        return username + returnStatement;
+    }
+
+    private String checkNumberOfClaims(Calendar calendarStartYear, Calendar calendarEndYear, List<Calendar> calendarClaimDates, List<String> listOfClaimStatuses) {
+        int claimCounter = 0;
+
+        for (int i = 0; i < calendarClaimDates.size(); i++) {
+            if (calendarClaimDates.get(i).after(calendarStartYear) && calendarClaimDates.get(i).before(calendarEndYear)) {

```

Comment:
You should move calendarClaimDates.get(i) to a local variable in loop scope so this is easier to read